nectar/src/command/trade/event_loop.rs

```diff
@@ -87,97 +87,95 @@ impl EventLoop {
                     }
                 },
                 event = self.swarm.next().fuse() => {
-                    self.handle_network_event(
-                            event,
-                        ).await?;
+                    let _ = self.handle_network_event(
+                        event,
+                    ).await.map_err(|err| tracing::error!("Network event handling failed: {:#}", err));
```

Comment:
This use of the `map_err` combinator is a bit odd because you are causing a side-effect (logging). Combinators are usually (pure) data pipelines but in this case your are just throwing away the result.

An alternative version of this could be:

```suggestion
                    if let Err(e) = self.handle_network_event(event).await {
                    	tracing::error!("Network event handling failed: {:#}", err));
                    }
```

Reply:
Interesting, I previously suggested the exact opposite to @luckysori because it was more compact :)

Reply:
Yeah I saw this pattern a number of times in nectar and always found it a bit irritating to be honest because we are not really mapping the error so it feels like an abuse of the combinator :/

I agree though that the alternative is also not really nice. Not sure how to solve this problem in a nice way :D

Reply:
I would try to avoid getting irritated when reading code 😉